### ISSUE TITLE
Support for .NET 6 for windows/linux web app

### ISSUE THREAD

**Issue by centreboard** [hashicorp/terraform-provider-azurerm#14297]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add .NET 6 as an option for `dotnet_version` on beta web app resources.
Currently errors
> Error: expected site_config.0.application_stack.0.dotnet_version to be one of [2.1 3.1 5.0], got 6.0

.NET 6 is allowed for existing `azurerm_app_service` as [dotnet_framework_version](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/app_service#dotnet_framework_version)

### New or Affected Resource(s)

* azurerm_windows_web_app
* azurerm_linux_web_app

### Potential Terraform Configuration

```hcl
resource "azurerm_linux_web_app" "example" {
  site_config {
    application_stack {
      dotnet_version = "6.0"
    }
  }
  ...
```

### References

* [.NET 6 on App Service](https://azure.github.io/AppService/2021/11/08/Dot.Net6.on.App.Service.html)
* `az webapp list-runtimes --linux` now lists .NET 6 and it's the default from the portal UI.
```
[
  "DOTNETCORE|2.1",
  "DOTNETCORE|3.1",
  "DOTNET|5.0",
  "DOTNET|6.0",
  ...
  ```

**Comment by jackofallops** [comment]
Thanks @centreboard - I have a local branch that includes this since the GA announcement, it's behind a few other changes, so will be along soon! :)

**Comment by surlypants** [comment]
any ETA when this might be released?

**Comment by surlypants** [comment]
turns out this is achievable today. M$ just has faulty documentation.

```az webapp list-runtimes --linux | grep DOTNET
  "DOTNETCORE|2.1",
  "DOTNETCORE|3.1",
  "DOTNET|5.0",
  "DOTNET|6.0",
```

5 & 6 _should_ be DOTNETCORE. supply (eg) "DOTNETCORE|6.0" to linux_fx_version and all's well.

**Comment by jackofallops** [comment]
@surlypants - fwiw, that list from `az webapp list-runtimes` isn't pulled from an API, iirc it's hard coded into the azure-cli tool.  The beta resources (available to try now) that will be released in 3.0 solve this by schema, rather than users having to work out arbitrary "fx" strings.

Since this seems to be resolved, I'm going to close it out for now.
